### PR TITLE
Shell log improvements

### PR DESCRIPTION
**Pull request by nordic-krch**
Shell log backend was enabled to early, before shell thread was up and running. That could lead to failure if log thread passes log message to shell log backend.
    
Additionally, modified algorithm for multiplexing log output with shell prompt. In case of logs flood prompt was unreadable because it was deleted immediately after being printed. Added k_sleep after printing prompt if any character was typed. This ensures that shell prompt is readable if user uses it.

Fixed the case when TXDONE signal was not cleared. Bug was unnoticed because UART backend was synchronous.

### CODE REVIEW

File: subsys/shell/shell.c [hunk 1]
```diff
@@ -1163,6 +1163,13 @@ static void shell_log_process(const struct shell *shell)
 		processed = shell_log_backend_process(shell->log_backend);
 		shell_current_command_print(shell);
 
+		/* Arbitrary delay added to ensure that prompt is readable and
+		 * can be used to enter further commands.
+		 */
+		if (shell->ctx->cmd_buff_len) {
+			k_sleep(15);
```
Comment (jukkar) [inline subsys/shell/shell.c]: It would be more readable to do `k_sleep(K_MSEC(15))` here instead

File: subsys/shell/shell.c [hunk 2]
```diff
@@ -97,6 +97,19 @@ static void shell_cmd_buffer_clear(const struct shell *shell)
...
 }
 
+static void shell_pend_on_txdone(const struct shell *shell)
+{
+	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
+		k_poll(&shell->ctx->events[SHELL_SIGNAL_TXDONE], 1, K_FOREVER);
+		k_poll_signal_reset(&shell->ctx->signals[SHELL_SIGNAL_TXDONE]);
+	} else {
+		/* Blocking wait in case of bare metal. */
+		while (!shell->ctx->internal.flags.tx_rdy) {
```
Comment (jukkar) [inline subsys/shell/shell.c]: This will spin like crazy, should we call k_yield() inside the while loop?

Comment (nordic-krch) [reply subsys/shell/shell.c]: it as case when MULTITHREADING is disabled so k_yield() is not an option. Do we have some sleep function in kernel?

File: subsys/shell/shell.c (hunk 1 above)
Comment (nordic-krch) [reply subsys/shell/shell.c]: done

File: subsys/shell/shell.c [hunk 3]
```diff
@@ -1163,6 +1163,13 @@ static void shell_log_process(const struct shell *shell)
 		processed = shell_log_backend_process(shell->log_backend);
```
Comment (jukkar) [inline subsys/shell/shell.c]: Noticed some typos in commit "shell: Postpone enabling log backend" message. 

s/was enabled to early/was enabled **too** early/

Comment (nordic-krch) [reply subsys/shell/shell.c]: fixed

**Review by jukkar — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**